### PR TITLE
Implement movie info component

### PR DESCRIPTION
**Pull request by KozlovaAngelika**
Implemented movie info component

### CODE REVIEW

File: src/components/MovieTile/MovieInfo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        }}
+        buttonStyle={styles.closeBtn}
+        containerStyle={styles.btnContainer}
+        onPress={closeModal}
+      />
+      <ScrollView showsVerticalScrollIndicator={false}>
+        <Card containerStyle={styles.card}>
+          <Card.Title>{data.title}</Card.Title>
+          <Text style={styles.description}>
+            У сотрудника крупного банка всё идёт по накатанной, пока однажды он
```
Comment (brusnev-egor) [inline src/components/MovieTile/MovieInfo/index.tsx:37]: remove hardcoded text

File: src/components/MovieTile/MovieInfo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import { Text, View, ScrollView } from 'react-native';
+import { Button, Overlay, Card } from 'react-native-elements';
+import { useTranslation } from 'react-i18next';
+import styles from './styles';
+import { COLORS } from '../../../utils/constants';
+import '../../../utils/i18n';
+
+interface Props {
+  isModalVisible: boolean;
+  closeModal: () => void;
```
Comment (brusnev-egor) [inline src/components/MovieTile/MovieInfo/index.tsx:11]: rename to `onClose`

File: src/components/MovieTile/MovieInfo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Text, View, ScrollView } from 'react-native';
+import { Button, Overlay, Card } from 'react-native-elements';
+import { useTranslation } from 'react-i18next';
+import styles from './styles';
+import { COLORS } from '../../../utils/constants';
+import '../../../utils/i18n';
+
+interface Props {
+  isModalVisible: boolean;
```
Comment (brusnev-egor) [inline src/components/MovieTile/MovieInfo/index.tsx:10]: np: can be just `isVisible`

File: src/components/MovieTile/MovieInfo/index.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Text, View, ScrollView } from 'react-native';
+import { Button, Overlay, Card } from 'react-native-elements';
+import { useTranslation } from 'react-i18next';
+import styles from './styles';
+import { COLORS } from '../../../utils/constants';
+import '../../../utils/i18n';
```
Comment (brusnev-egor) [inline src/components/MovieTile/MovieInfo/index.tsx:7]: unused import

File: src/utils/i18n/locales/en.json [hunk 5]
```diff
@@ -2,5 +2,7 @@
   "home": "Home",
   "top250": "Top 250",
   "favorites": "Favorites",
-  "search": "Search"
+  "search": "Search",
+  "addtofavorites": "Add to favorites",
```
Comment (brusnev-egor) [inline src/utils/i18n/locales/en.json:6]: camelcase or snake_case here

File: src/utils/i18n/locales/ru.json [hunk 6]
```diff
@@ -2,5 +2,7 @@
   "home": "Главная",
   "top250": "Топ 250",
   "favorites": "Избранное",
-  "search": "Поиск"
+  "search": "Поиск",
+  "addtofavorites": "Добавить в избранное",
```
Comment (brusnev-egor) [inline src/utils/i18n/locales/ru.json:6]: camelcase or snake_case here

File: src/components/MovieTile/index.tsx [hunk 7]
```diff
@@ -1,21 +1,40 @@
...
-      }}
-      resizeMode="contain"
-    />
-  </Card>
-);
+const MovieTile: React.FC<Props> = ({ data }) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const toggleModal = (isVisibleModal: boolean): void => {
+    setIsVisible(!isVisibleModal);
+  };
```
Comment (i-nemets) [inline src/components/MovieTile/index.tsx:13]: i think, better use ` setIsVisible( isVisibleModal => !isVisibleModal)`

File: src/components/MovieTile/index.tsx [hunk 8]
```diff
@@ -1,36 +1,55 @@
...
 import { Button, Card } from 'react-native-elements';
 import { COLORS } from '../../utils/constants';
+import MovieInfo from './MovieInfo';
 import styles from './styles';
 
 interface Props {
-  title: string;
-  imgSrc: string;
+  data: Movie;
   isDeleteBtn: boolean;
```
Comment (brusnev-egor) [inline src/components/MovieTile/index.tsx:9]: rename prop to `isInFavorites` for example

File: src/components/MovieTile/index.tsx [hunk 9]
```diff
@@ -1,36 +1,55 @@
...
+        onClose={() => {
+          toggleModal(isVisible);
+        }}
+        data={data}
       />
-    ) : null}
-  </Card>
-);
+      {isDeleteBtn ? (
+        <Button
```
Comment (brusnev-egor) [inline src/components/MovieTile/index.tsx:40]: as I remember this button should also add movie to favorites, not only delete

File: src/components/MovieTile/MovieInfo/styles.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  ratingContainer: {
+    flexDirection: 'row',
+    padding: 10,
+  },
+  ratingTitle: {
+    marginRight: 8,
+  },
+  btnContainer: {
+    flexDirection: 'row-reverse',
```
Comment (brusnev-egor) [inline src/components/MovieTile/MovieInfo/styles.ts:31]: why do you need row reverse here?

File: src/components/MovieTile/index.tsx [hunk 11]
```diff
@@ -1,36 +1,54 @@
...
-    />
-    {isDeleteBtn ? (
-      <Button
-        icon={{
-          name: 'delete',
-          size: 20,
-          color: COLORS.LIGHT_GREY,
+const MovieTile: React.FC<Props> = ({ data, isDeleteBtn }) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const toggleModal = (): void =>
```
Comment (brusnev-egor) [inline src/components/MovieTile/index.tsx:14]: surround it with braces

File: src/components/MovieTile/index.tsx [hunk 12]
```diff
@@ -1,36 +1,54 @@
...
+    setIsVisible((isVisibleModal: boolean) => !isVisibleModal);
+  return (
+    <Card>
+      <Card.Title
+        onPress={() => {
+          toggleModal();
```
Comment (brusnev-egor) [inline src/components/MovieTile/index.tsx:20]: pass pointer instead of arrow function

File: src/components/MovieTile/MovieInfo/index.tsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  onClose: () => void;
+  data: Movie;
+}
+
+const MovieInfo: React.FC<Props> = ({ isVisible, onClose, data }) => {
+  const { t } = useTranslation();
+  return (
+    <Overlay isVisible={isVisible} fullScreen overlayStyle={styles.overlay}>
+      <Button
+        icon={{
```
Comment (brusnev-egor) [inline src/components/MovieTile/MovieInfo/index.tsx:19]: a lot of duplication of this one. please add ThemeProvider

Comment (brusnev-egor) [reply src/components/MovieTile/MovieInfo/index.tsx:19]: https://reactnativeelements.com/docs/customization/#using-themeprovider

File: src/theme.ts [hunk 14]
```diff
@@ -1,2 +1,14 @@
...
+    titleStyle: {
+      color: COLORS.WHITE,
+    },
+    buttonStyle: {
+      backgroundColor: COLORS.LIGHT_GREY,
+      borderRadius: 10,
+    },
+  },
+};
 export default theme;
```
Comment (i-nemets) [inline src/theme.ts]: leave one empty line before export

File: src/components/MovieTile/MovieInfo/index.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+            <Text style={styles.ratingTitle}>{t('rating')}</Text>
+            <Text>{data.imDbRating}</Text>
+          </View>
+          <Text>{data.description}</Text>
+          <Card.Image
+            source={{
+              uri: data.image,
+            }}
+            resizeMode="contain"
+          />
```
Comment (i-nemets) [inline src/components/MovieTile/MovieInfo/index.tsx:42]: I think it's better to write this on one line for readability.
` <Card.Image source={{uri: data.image}} resizeMode="contain" />`

File: src/components/MovieTile/index.tsx [hunk 16]
```diff
@@ -1,36 +1,43 @@
...
-      <Button
-        icon={{
-          name: 'delete',
-          size: 20,
-          color: COLORS.LIGHT_GREY,
+const MovieTile: React.FC<Props> = ({ data, isInFavorites }) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const toggleModal = (): void => {
+    setIsVisible((isVisibleModal: boolean) => !isVisibleModal);
+  };
```
Comment (i-nemets) [inline src/components/MovieTile/index.tsx:16]: i think, better use `setIsVisible( isVisibleModal => !isVisibleModal)`

**Review by brusnev-egor — APPROVED** [review]

**Review by i-nemets — APPROVED** [review]

**Pull request merged**